modules/lang-painless/src/main/java/org/elasticsearch/painless/DefMath.java

```diff
@@ -561,7 +565,7 @@ private static boolean eq(Object left, Object right) {
                 if (left instanceof Number) {
                     return ((Number) left).longValue() == (long) right;
                 } else if (left instanceof Character) {
-                    return (char) left == ((Number) right).longValue();
+                    return (char) left == (long) right;
```

Comment:
We're also missing a test that hits this code, can you add one?

Reply:
Added